src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -240,22 +261,8 @@ public TopicPartitionChannel(
     final long lastCommittedOffsetToken = fetchOffsetTokenWithRetry();
     this.offsetPersistedInSnowflake.set(lastCommittedOffsetToken);
     this.processedOffset.set(lastCommittedOffsetToken);
-    // Reset the consumer offset in kafka, only if we have a valid offset token at server side. It's
-    // possible that we hit the invalid offset exception if KC stops and the data expired, so we
-    // will just accept what kafka sends us in this case
     if (lastCommittedOffsetToken != NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE) {
-      try {
-        this.sinkTaskContext.offset(this.topicPartition, lastCommittedOffsetToken);
-      } catch (InvalidOffsetException exception) {
-        LOGGER.error(
-            "Invalid offset exception encountered during resetting the offset for topic"
-                + " partition:{}, channel:{}, offset:{}, we will just accept what kafka sends us",
-            this.topicPartition.toString(),
-            this.channel.getFullyQualifiedName(),
-            lastCommittedOffsetToken);
-        this.offsetPersistedInSnowflake.set(NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE);
-        this.processedOffset.set(NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE);
-      }
+      this.sinkTaskContext.offset(this.topicPartition, lastCommittedOffsetToken);
```

Comment:
good catch, updated! I don't think it affects correctness though because the first record will always be skipped based on the logic in `insertRecordToBuffer`

Reply:
yes correct, I also tested it out with + 1 and this code and behavior was same as you mentioned. But it is better to be consistent? Also less prone to exceptions if ever customers to decide remove offsets till offsetToken in their snowflake channel.